src/features/unified-share-modal/flowTypes.js

```diff
@@ -171,3 +172,128 @@ export type tooltipComponentIdentifierType =
+    /** Handler function for clicks on the settings icon. If not provided, the settings icon won't be rendered. */
+    onSettingsClick?: Function,
+    /** Shows a callout tooltip next to the names / email addresses input field explaining pre-populated recommendation */
+    recommendedSharingTooltipCalloutName: ?string,
+    /**
+     * Function to send collab invitations based on the given parameters object.
+     * This function should return a Promise.
+     */
+    sendInvites: (params: Object) => Promise<Object>,
+    /** Message indicating an error occurred while sending the invites. */
+    sendInvitesError: React.Node,
+    /**
+     * Function to send shared link email based on the given parameters object.
+     * This function should return a Promise.
+     */
+    sendSharedLink: (params: Object) => Promise<Object>,
+    /** Message indicating an error occurred while sending the shared link. */
+    sendSharedLinkError: React.Node,
+    /** Function hoists contact data upon updates to the parent component. Only needed for suggested collabs. */
+    setUpdatedContacts?: (inviteCollabsContacts: Array<contactType>) => void,
+    /** Shared link data */
+    sharedLink: sharedLinkType,
+    /** Determine whether to show the First-time experience tooltip on load */
+    showCalloutForUser?: boolean,
+    /** Shows a callout tooltip next to the names / email addresses input field encouraging users to fill out coworkers contact info */
+    showEnterEmailsCallout?: boolean,
+    /** Whether only the USF should be rendered */
+    showFormOnly?: boolean,
+    /** Shows a callout tooltip next gear icon with info about what can be customized */
+    showSharedLinkSettingsCallout?: boolean,
+    /**
+     * Flag to show link to upgrade and get more access controls.
+     * Only applicable to non-file item types.
+     */
+    showUpgradeOptions: boolean,
+    /** Whether or not a request is in progress */
+    submitting: boolean,
+    /** Data for suggested collaborators shown at bottom of input box. UI doesn't render when this has length of 0. */
+    suggestedCollaborators?: SuggestedCollabLookup,
+    /** Mapping of components to the content that should be rendered in their tooltips */
+    tooltips?: { [componentIdentifier: tooltipComponentIdentifierType]: React.Node },
+    /** Object with props and handlers for tracking interactions */
+    trackingProps: trackingPropsType,
+};
+
+// Prop types for the Unified Share Modal
+export type USMProps = BaseUSMProps & {
+    /** Whether the USM is open */
+    isOpen?: boolean,
+    /** ID for the item being shared - for use with V2 APIs only */
+    itemID?: string,
+    /** Type of the item being shared - for use with V2 APIs only */
+    itemType?: string,
+    /** Language for the USM/USF */
+    language?: string,
+    /** Handler function for when the USM is closed */
+    onRequestClose?: Function,
+    /** Access token - for use with V2 APIs only */
+    token?: string,
```

Comment:
would this need to be passed in? I would assume the wrapper component would handle all of the logic around APIs and tokens

Reply:
Yes, good catch. I removed the API-specific props. Can you review this PR again?